jdaviz/configs/cubeviz/plugins/tests/test_cubeviz_aperphot.py

```diff
@@ -132,21 +149,49 @@ def test_cubeviz_aperphot_cube_orig_flux_mjysr(cubeviz_helper, spectrum1d_cube_c
     plg.aperture_selected = "Subset 1"
     plg.background_selected = "Subset 2"
 
-    # Make sure per steradian is handled properly.
-    assert_allclose(plg.pixel_area, 0.01)
-    assert_allclose(plg.flux_scaling, 0.003631)
+    #  Check that the default flux scaling is present for MJy / sr cubes
+    if cube_unit == (u.MJy / u.sr):
+        assert_allclose(plg.flux_scaling, 0.003631)
+    else:
+        assert_allclose(plg.flux_scaling, 0.0)
+
+    # if cube is MJy / sr, set pixel area to 1 sr / pix2 so
+    # we can directly compare outputs between per sr and per pixel cubes, which
+    # will give the same results with this scaling
+    if cube_unit == (u.MJy / u.sr):
+        assert_allclose(plg.pixel_area, 0.01)  # check default
+        plg.pixel_area = 1 * (u.sr).to(u.arcsec*u.arcsec)
+        solid_angle_unit = u.sr
+        assert not plg.disable_pixarea_input  # for /sr cubes, this should be editable
+
+    else:
+        # for per pixel cubes, set flux scaling to default for MJy / sr cubes
+        # so we can directly compare. this shouldn't be populated automatically,
+        # which is checked above
+        plg.flux_scaling = 0.003631
+        solid_angle_unit = u.pix * u.pix
+        cube_unit = u.MJy / solid_angle_unit  # cube unit in app is now per pix2
+        assert plg.disable_pixarea_input
 
     plg.vue_do_aper_phot()
     row = cubeviz_helper.get_aperture_photometry_results()[0]
 
     # Basically, we should recover the input rectangle here, minus background.
     assert_allclose(row["xcenter"], 3 * u.pix)
     assert_allclose(row["ycenter"], 1 * u.pix)
-    assert_allclose(row["sum"], 1.1752215e-12 * u.MJy)  # (15 - 10) MJy/sr x 2.3504431e-13 sr
+    # (15 - 10) MJy/sr x 1 sr, will always be MJy since solid angle is multiplied out
+    assert_allclose(row["sum"], 5.0 * u.MJy)
```

Comment:
What happened that caused such a drastic change in expected output in this test?

Reply:
I changed the scale factor so that i could directly compare pix2 and sr, so the results changed intentionally

Reply:
sure. also lets move over to my PR for discussion since these changes were made there?

Reply:
changed this in 3156